File: src/test/java/io/icednut/algorithm/exercise/etc/leet1684/SolutionTest.java
package io.icednut.algorithm.exercise.etc.leet1684;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class SolutionTest {

    @Test
    void test01() {
        Solution solution = new Solution();

        int output = solution.countConsistentStrings("ab", new String[]{"ad", "bd", "aaab", "baa", "badab"});

        assertEquals(2, output);
    }

    @Test
    void test02() {
        Solution solution = new Solution();

        int output = solution.countConsistentStrings("abc", new String[]{"a","b","c","ab","ac","bc","abc"});

        assertEquals(7, output);
    }
}

File: src/test/kotlin/io/icednut/algorithm/exercise/greedy/leetcode1578/SolutionTest.kt
package io.icednut.algorithm.exercise.greedy.leetcode1578

import org.junit.jupiter.api.Assertions.*
import kotlin.test.Test

class SolutionTest {

    @Test
    fun test01() {
        val solution = Solution()

        // [259,770],[448,54],[926,667],[184,139],[840,118],[577,469]
        val result = solution.twoCitySchedCost(
            arrayOf(
                intArrayOf(259, 770),
                intArrayOf(448, 54),
                intArrayOf(926, 667),
                intArrayOf(184, 139),
                intArrayOf(840, 118),
                intArrayOf(577, 469),
            )
        )

        assertEquals(1859, result)
    }
}